emission/storage/decorations/trip_queries.py

```diff
@@ -162,17 +188,24 @@ def get_not_deleted_candidates(filter_fn, potential_candidates):
         return []
 
     # We want to retain all ACTIVE entries that have not been DELETED
-    all_active_list = [efpc for efpc in extra_filtered_potential_candidates if "status" not in efpc.data or efpc.data.status == ecwtui.InputStatus.ACTIVE]
-    all_deleted_id = [efpc["data"]["match_id"] for efpc in extra_filtered_potential_candidates if "status" in efpc.data and efpc.data.status == ecwtui.InputStatus.DELETED]
+    all_active_list = [efpc for efpc in extra_filtered_potential_candidates if "status" not in efpc.data or efpc.data.status == ecwui.InputStatus.ACTIVE]
+    all_deleted_id = [efpc["data"]["match_id"] for efpc in extra_filtered_potential_candidates if "status" in efpc.data and efpc.data.status == ecwui.InputStatus.DELETED]
     # TODO: Replace this with filter and a lambda if we decide not to match by ID after all
     not_deleted_active = [efpc for efpc in all_active_list if efpc["data"]["match_id"] not in all_deleted_id]
     logging.info(f"Found {len(all_active_list)} active entries, {len(all_deleted_id)} deleted entries -> {len(not_deleted_active)} non deleted active entries")
     return not_deleted_active
 
-def get_user_input_for_trip_object(ts, trip_obj, user_input_key):
-    tq = estt.TimeQuery("data.start_ts", trip_obj.data.start_ts, trip_obj.data.end_ts)
+def get_user_input_for_timeline_entry_object(ts, timeline_entry, user_input_key):
+    # timeline_entry can be either a trip or place, so let's check for start/enter or end/exit
+    if (hasattr(timeline_entry.data, 'start_ts')):
+        start = timeline_entry.data.start_ts
+        end = timeline_entry.data.end_ts
+    else:
+        start = timeline_entry.data.enter_ts
+        end = timeline_entry.data.exit_ts
```

Comment:
this code again seems to be duplicated

Reply:
I'll refactor into helper functions 'begin' and 'stop'